iroha_http_server/src/lib.rs

```diff
@@ -521,27 +582,115 @@ pub struct RouteBuilder<'s, State> {
     server: &'s mut Server<State>,
 }
 
+async fn wrapper_handler<State, Path, Query, Req, Fut, F>(
+    fut: F,
+    state: State,
+    path: Result<Path, Path::Error>,
+    query: Result<Query, Query::Error>,
+    req: Result<Req, Req::Error>,
+) -> http::HttpResponse
+where
+    State: Clone + Send + Sync + 'static,
+    Fut: Future<Output = http::HttpResponse>,
+    F: Send + Sync + 'static + Fn(State, Path, Query, Req) -> Fut,
+    Path: TryFrom<http::PathParams> + Send + Sync + 'static,
+    Path::Error: http::HttpResponseError,
+    Query: TryFrom<http::QueryParams> + Send + Sync + 'static,
+    Query::Error: http::HttpResponseError,
+    Req: TryFrom<http::HttpRequest> + Send + Sync + 'static,
+    Req::Error: http::HttpResponseError,
+{
+    let path = match path {
+        Ok(path) => path,
+        Err(err) => return http::HttpResponseError::error_response(&err),
+    };
+    let query = match query {
+        Ok(query) => query,
+        Err(err) => return http::HttpResponseError::error_response(&err),
+    };
+    let req = match req {
+        Ok(req) => req,
+        Err(err) => return http::HttpResponseError::error_response(&err),
+    };
+    fut(state, path, query, req).await
+}
+
 impl<'s, State> RouteBuilder<'s, State>
 where
     State: Clone + Send + Sync + 'static,
 {
     /// Add GET handler at the specified url.
-    pub fn get(&mut self, handler: impl HttpHandler<State>) {
+    pub fn get<Path, Query, Req, F, Fut>(&mut self, handler: F)
```

Comment:
Why would you like to remove `HttpHandler` trait and instead use raw function here? From what I remember `HttpHandler` is implemented for functions. Also you left this trait in the codebase, but it seems it is not used anymore.

Reply:
The usage of the trait will also reduce the number of biolerplate generic bounds that we have here in several functions.

Reply:
It is not possible to constrain F using trait, but it is possible in methods. Having trait might not be necessary, you are right

Reply:
Trait can be implemented for a specific constrained F like it is done there, or am I missing something?

Reply:
It can be, but Fn trait can be implemented for any type in nightly. This creates constraints.